### DISCUSSION TITLE
callback function with numpy parameter problem

### DISCUSSION THREAD

**Discussion by Fokatu** [pybind/pybind11#3172] [Q&A]
two problems.

1. what wrong with the following code?  When the function is called, I got an exception 
![image](https://user-images.githubusercontent.com/25919117/128167669-ccf90cf8-a6cb-4441-8361-2df2a5134700.png)

```
void test()
{
	double* data = new double[3 * 5];
	py::array_t<double> arr(3 * 5, data);
	delete[] data;
}
```

2. My need is to create a python callback function with a numpy ndarray parameter. But when I try to pack the data as a py::array_t object in cpp code I got the problem in 1. Is there another way to pass double[] to python callback? 
 
Thanks!

**Comment by henryiii** [comment]
Please use pybind11 not the C api. `py::module_::import(name)`, in this case. I'm not an expert in this area, but pretty sure there's more to it than this (look at the implementation of `py::module_::import`).
